horace_core/sqw/page_operations/PageOp_join_sqw.m

```diff
@@ -0,0 +1,168 @@
+            % Overload of split method allowing to define large target chink
+            % and store npix_idx for internal usage
+            % Inputs:
+            % npix  -- image npix array, which defines the number of pixels
+            %           contributing into each image bin and the pixels
+            %           ordering in the linear array
+            % chunk_size
+            %       -- sized of chunks to split pixels
+            % Returns:
+            % npix_chunks -- cellarray, containing the npix parts
+            % npix_idx    -- [2,n_chunks] array of indices of the chunks in
+            %                the npix array.
+            % See split procedure for more details
+            fb = config_store.instance().get_value( ...
+                'hor_config','fb_scale_factor');
+            % do large chunk to decrease number of sub-calls to each data
+            % pixels
+            large_chunk = chunk_size*fb;
+            [npix_chunks, npix_idx,obj] = split_into_pages@PageOpBase(obj,npix,large_chunk);
+            obj.block_idx_ = npix_idx;
+            page_sizes = cellfun(@(x)sum(x(:)),npix_chunks);
+            page_pos = cumsum(page_sizes);
+            obj.page_start_pos_ = [1,page_pos(1:end-1)];
+        end
+
+        function obj = get_page_data(obj,idx,npix_blocks)
+            % join-specific access to block of page data
+            %
+            % reads data from mutiple sources and combines them together
+            % into single page of data.
+            %
+            bin_start = cumsum(npix_blocks{idx});
+            page_size = bin_start(end);
+            % the positions of empty bins to place pixels
+            bin_start = [0,bin_start(1:end-1)] + 1;
+            page_data = zeros(PixelDataBase.DEFAULT_NUM_PIX_FIELDS,page_size);
+            multi_data   = obj.pix_combine_info;
+            n_datasets = multi_data.nfiles;
+            for i=1:n_datasets
+                % get particular dataset's page data
+                [contr_page_data,page_bin_distr] = multi_data.get_dataset_page( ...
+                    i,obj.current_page_pix_pos_(i),obj.block_idx_(:,idx));
+                % advance initial page position for the particular dataset
+                n_page_pix = size(contr_page_data,2);
+                obj.current_page_pix_pos_(i) = ...
+                    obj.current_page_pix_pos_(i)+n_page_pix;
+                %
+                if ~isempty(obj.new_runid)
+                    contr_page_data(obj.run_idx_,:) = obj.new_runid(i);
+                end
+
+                % find indexes of i-th dataset's page pixels in the target's
+                % dataset page
+                targ_bin_idx = fill_idx(bin_start,page_bin_distr);
+                % targ_bin_pos   = repelem(bin_start,page_bin_distr);
+                % cell_idx = arrayfun(@fill_fun,page_bin_distr,'UniformOutput',false);
+                % targ_bin_idx   = targ_bin_pos+[cell_idx{:}]';
```

Comment:
Debugging statements?

Reply:
Will be checked tomorrow. Letf for profiling